Unable to build with cross-compiler using installed library

**Describe the bug**
I'm unable to configure a project when cross-compiling with an installed version of the library.

```
# cmake -G Ninja -S /work/minimal/ -B /work/minimal/build-mingw-x64-release \
  -DCMAKE_PREFIX_PATH=/work/depends/install/cross-x64 \
  -DCMAKE_TOOLCHAIN_FILE=/work/llvm-mingw.cmake
-- The CXX compiler identification is Clang 22.1.4
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/llvm-mingw/bin/x86_64-w64-mingw32-clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.2s)
CMake Error at /work/depends/install/cross-x64/lib/cmake/hpp_proto/protobuf_generate_hpp.cmake:158 (add_custom_command):
  Error evaluating generator expression:

    $<TARGET_FILE:hpp_proto::protoc-gen-hpp>

  No target "hpp_proto::protoc-gen-hpp"
Call Stack (most recent call first):
  /work/depends/install/cross-x64/lib/cmake/hpp_proto/protobuf_generate_hpp.cmake:180 (protobuf_generate)
  CMakeLists.txt:12 (protobuf_generate_hpp)


CMake Error at /work/depends/install/cross-x64/lib/cmake/hpp_proto/protobuf_generate_hpp.cmake:158 (add_custom_command):
  Error evaluating generator expression:

    $<TARGET_FILE:hpp_proto::protoc-gen-hpp>

  No target "hpp_proto::protoc-gen-hpp"
Call Stack (most recent call first):
  /work/depends/install/cross-x64/lib/cmake/hpp_proto/protobuf_generate_hpp.cmake:180 (protobuf_generate)
  CMakeLists.txt:12 (protobuf_generate_hpp)


CMake Error at /work/depends/install/cross-x64/lib/cmake/hpp_proto/protobuf_generate_hpp.cmake:158 (add_custom_command):
  Error evaluating generator expression:

    $<TARGET_FILE:hpp_proto::protoc-gen-hpp>

  No target "hpp_proto::protoc-gen-hpp"
Call Stack (most recent call first):
  /work/depends/install/cross-x64/lib/cmake/hpp_proto/protobuf_generate_hpp.cmake:180 (protobuf_generate)
  CMakeLists.txt:12 (protobuf_generate_hpp)


-- Generating done (0.0s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```
Full log: https://github.com/llm96/hpp-proto-system-cross-test/actions/runs/25331028713/job/74264486793

Additional bug where `IS_EXECUTABLE` is not usable with CMake 3.28. I think this was introduced with 3.29.
https://discourse.cmake.org/t/cmake-3-29-0-rc3-is-ready-for-testing/10283#p-29671-commands-6

```
CMake Version: 3.28.3
-- The CXX compiler identification is Clang 22.1.4
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/llvm-mingw/bin/x86_64-w64-mingw32-clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:86 (if):
  if given arguments:

    "NOT" "IS_EXECUTABLE" "/__w/hpp-proto-system-cross-test/hpp-proto-system-cross-test/depends/install/host/bin/protoc-gen-hpp"

  Unknown arguments specified

```

**To Reproduce**
Steps to reproduce the behavior:
1. Install host version of the library
2. Install cross version of the library (with `-DHPP_PROTO_PROTOC_GEN_HPP` set to host `protoc-gen-hpp`)
3. Try to configure a project with cross toolchain

Full reproduction script: https://github.com/llm96/hpp-proto-system-cross-test/blob/master/build.sh

**Expected behavior**
Project should configure and build without error.

**Screenshots**
N/A

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04 (x86_64 & AArch64)
 - Compiler: Clang 22.1.4
 - CMake Version: 3.28.3, 4.3.2
 - Protobuf Version: 34.1

**Additional context**
Workaround patch: https://github.com/llm96/hpp-proto-system-cross-test/blob/master/fix-cross-config.diff

--- a/.github/workflows/ubuntu-two-stage-host-protoc-gen-hpp.yml
+++ b/.github/workflows/ubuntu-two-stage-host-protoc-gen-hpp.yml
@@ -22,6 +22,10 @@ env:
 jobs:
   build_and_test_with_host_plugin:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake: ["3.28.3", "3.31.0"]
     steps:
       - uses: actions/checkout@v6
 
@@ -29,7 +33,7 @@ jobs:
         uses: aminya/setup-cpp@v1.8.0
         with:
           compiler: gcc-13
-          cmake: "3.31.0"
+          cmake: ${{ matrix.cmake }}
           ninja: true
 
       - name: Install protobuf/grpc tools
@@ -58,3 +62,43 @@ jobs:
 
       - name: Test
         run: ctest --test-dir build --output-on-failure
+
+      - name: Install external-plugin build
+        run: |
+          cmake --install build --prefix "${{ github.workspace }}/install-cross"
+
+      - name: Write cross consumer fixture
+        run: |
+          mkdir -p cross-consumer
+          cat > cross-consumer/CMakeLists.txt <<'EOF'
+          cmake_minimum_required(VERSION 3.25)
+          project(hpp_proto_cross_consumer LANGUAGES CXX)
+
+          find_package(hpp_proto CONFIG REQUIRED)
+
+          add_library(cross_consumer_messages INTERFACE)
+          protobuf_generate_hpp(
+            TARGET cross_consumer_messages
+            PROTOS "${CMAKE_CURRENT_SOURCE_DIR}/message.proto"
+            PROTOC_OUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
+          EOF
+          cat > cross-consumer/message.proto <<'EOF'
+          syntax = "proto3";
+          package hpp_proto_cross_consumer;
+
+          message Smoke {
+            string value = 1;
+          }
+          EOF
+          cat > cross-toolchain.cmake <<'EOF'
+          set(CMAKE_SYSTEM_NAME Generic)
+          EOF
+
+      - name: Configure installed package from cross consumer
+        run: |
+          cmake -S cross-consumer -B cross-consumer-build -G Ninja \
+            -DCMAKE_PREFIX_PATH="${{ github.workspace }}/install-cross" \
+            -DCMAKE_TOOLCHAIN_FILE="${{ github.workspace }}/cross-toolchain.cmake"
+
+      - name: Build cross consumer
+        run: cmake --build cross-consumer-build -j2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,9 +83,15 @@ if(HPP_PROTO_PROTOC_GEN_HPP)
     message(FATAL_ERROR
       "HPP_PROTO_PROTOC_GEN_HPP does not exist: ${HPP_PROTO_PROTOC_GEN_HPP}")
   endif()
-  if(NOT IS_EXECUTABLE "${HPP_PROTO_PROTOC_GEN_HPP}")
+  if(IS_DIRECTORY "${HPP_PROTO_PROTOC_GEN_HPP}")
     message(FATAL_ERROR
-      "HPP_PROTO_PROTOC_GEN_HPP is not executable: ${HPP_PROTO_PROTOC_GEN_HPP}")
+      "HPP_PROTO_PROTOC_GEN_HPP must be a file: ${HPP_PROTO_PROTOC_GEN_HPP}")
+  endif()
+  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.29)
+    if(NOT IS_EXECUTABLE "${HPP_PROTO_PROTOC_GEN_HPP}")
+      message(FATAL_ERROR
+        "HPP_PROTO_PROTOC_GEN_HPP is not executable: ${HPP_PROTO_PROTOC_GEN_HPP}")
+    endif()
   endif()
 endif()
 

--- a/hpp_proto-config.cmake.in
+++ b/hpp_proto-config.cmake.in
@@ -3,6 +3,9 @@ include(CMakeFindDependencyMacro)
 
 get_filename_component(_hpp_proto_prefix "${CMAKE_CURRENT_LIST_DIR}/../../.." ABSOLUTE)
 set(HPP_PROTO_ROOT "${_hpp_proto_prefix}")
+if(NOT PROTOC_GEN_HPP_PROGRAM)
+    set(PROTOC_GEN_HPP_PROGRAM "@HPP_PROTO_PROTOC_GEN_HPP@")
+endif()
 
 find_dependency(glaze)
 if(NOT @is_utf8_ADDED@)
@@ -30,6 +33,18 @@ if(NOT TARGET hpp_proto::protoc)
     # Create executable imported target hpp_proto::protoc
     add_executable(hpp_proto::protoc IMPORTED)
     set_property(TARGET hpp_proto::protoc PROPERTY IMPORTED_LOCATION "${PROTOC_PROGRAM}")
+endif()
+
+if(NOT TARGET hpp_proto::protoc-gen-hpp)
+    if(NOT PROTOC_GEN_HPP_PROGRAM)
+        message(FATAL_ERROR "Could not find 'protoc-gen-hpp'. Set PROTOC_GEN_HPP_PROGRAM to an absolute path.")
+    endif()
+    get_filename_component(PROTOC_GEN_HPP_PROGRAM "${PROTOC_GEN_HPP_PROGRAM}" ABSOLUTE)
+    if(NOT EXISTS "${PROTOC_GEN_HPP_PROGRAM}" OR IS_DIRECTORY "${PROTOC_GEN_HPP_PROGRAM}")
+        message(FATAL_ERROR "PROTOC_GEN_HPP_PROGRAM must be an existing file: ${PROTOC_GEN_HPP_PROGRAM}")
+    endif()
+    add_executable(hpp_proto::protoc-gen-hpp IMPORTED)
+    set_property(TARGET hpp_proto::protoc-gen-hpp PROPERTY IMPORTED_LOCATION "${PROTOC_GEN_HPP_PROGRAM}")
 endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/protobuf_generate_hpp.cmake")

